implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -85,9 +85,12 @@ private void createDatabaseIfNecessary() {
                     .setRetentionReplicationFactor(config.retentionReplicationFactor())
                     .setRetentionShardDuration(config.retentionShardDuration()).build();
 
-            httpClient
+            HttpSender.Request.Builder requestBuilder = httpClient
                     .post(config.uri() + "/query?q=" + URLEncoder.encode(createDatabaseQuery, "UTF-8"))
-                    .withBasicAuthentication(config.userName(), config.password())
+                    .withBasicAuthentication(config.userName(), config.password());
+            config.apiVersion().addHeaderToken(config, requestBuilder);
```

Comment:
I was curious about this too in my testing. It will overwrite the `Authorization` header, which is the behavior we want, but we should make sure we are testing for it, to prevent any bugs from that behavior potentially changing in the future. I will add a note to the JavaDoc for the `userName` and `password` config methods to mention they won't be used if a token is configured.

> I would do this login this class not in the config.

Could you clarify what you mean?

Reply:
typo, I wanted to write this: I would do this logic in this class not in the config.